File: models/store/item.js
const { DataTypes } = require('sequelize');
const sequelize = require('../../config/database');

const Item = sequelize.define('Item', {
    id: {
        type: DataTypes.UUID,
        primaryKey: true,
        defaultValue: DataTypes.UUIDV4
    },
    institution_id: {
        type: DataTypes.UUID,
        allowNull: true
    },
    name: {
        type: DataTypes.STRING(255),
        allowNull: false,
    },
    description: {
        type: DataTypes.TEXT,
        allowNull: true,
    },
    category: {
        type: DataTypes.ENUM(
            'medicine',
            'medical_equipment',
            'surgical_supplies',
            'laboratory',
            'radiology',
            'consumables',
            'office_supplies',
            'cleaning_supplies'
        ),
        allowNull: false
    },
    unit_of_measure: {
        type: DataTypes.STRING(50),
        allowNull: false,
        defaultValue: 'pieces'
    },
    reorder_level: {
        type: DataTypes.INTEGER,
        allowNull: false,
        defaultValue: 10
    },
    critical_level: {
        type: DataTypes.INTEGER,
        allowNull: false,
        defaultValue: 5
    },
    is_active: {
        type: DataTypes.BOOLEAN,
        defaultValue: true
    }
}, {
    tableName: 'items',
    timestamps: true,
});

// Define associations
Item.associate = (models) => {
    Item.hasMany(models.ItemBatch, { foreignKey: 'item_id', as: 'batches' });
    Item.hasMany(models.StockAlert, { foreignKey: 'item_id', as: 'alerts' });
    Item.belongsTo(models.Supplier, { foreignKey: 'supplier_id', as: 'supplier' });
};

module.exports = Item;

File: store-service/app/models/item.py
"""Item model - stock items catalog."""

import uuid
from datetime import datetime

from sqlalchemy import Column, DateTime, Enum, Integer, String, Text, Boolean
from sqlalchemy.dialects.postgresql import UUID

from app.database import Base
from app.utils.types import ItemCategory


class Item(Base):
    __tablename__ = "items"

    id = Column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4)
    institution_id = Column(UUID(as_uuid=True), nullable=True)
    name = Column(String(255), nullable=False)
    description = Column(Text, nullable=True)
    category = Column(Enum(ItemCategory), nullable=False)
    unit_of_measure = Column(String(50), nullable=False, default="pieces")
    reorder_level = Column(Integer, nullable=False, default=10)
    critical_level = Column(Integer, nullable=False, default=5)
    supplier_id = Column(UUID(as_uuid=True), nullable=True)
    is_active = Column(Boolean, default=True)

    created_at = Column(DateTime(timezone=True), default=datetime.utcnow)
    updated_at = Column(DateTime(timezone=True), default=datetime.utcnow, onupdate=datetime.utcnow)
